[VL] compile fail on JDK11

### Backend

VL (Velox)

### Bug description

The method `newInstance()` seems deprecated and brings fatal warning. 

`mvn clean package -Pbackends-velox -Prss -Pspark-3.3 -DskipTests -Dspark.version=3.3.2`

```
[WARNING] /home/sparkuser/git/gluten/gluten-core/src/main/scala/io/glutenproject/execution/ScanTransformerFactory.scala:74: warning: method newInstance in class Class is deprecated
[WARNING]           .newInstance()
[WARNING]            ^
[ERROR] error: No warnings can be incurred under -Xfatal-warnings.
[WARNING] one warning found
[ERROR] one error found
[ERROR] exception compilation error occurred!!!
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:153)
    at scala_maven_executions.JavaMainCallerByFork.run (JavaMainCallerByFork.java:95)
    at scala_maven.ScalaCompilerSupport.compile (ScalaCompilerSupport.java:173)
    at scala_maven.ScalaCompilerSupport.doExecute (ScalaCompilerSupport.java:86)
    at scala_maven.ScalaMojoSupport.execute (ScalaMojoSupport.java:310)
    at scala_maven.ScalaCompileMojo.execute (ScalaCompileMojo.java:108)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] prepare-compile in 0.0 s
[INFO] compile in 18.1 s
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Gluten Parent Pom 1.1.0-SNAPSHOT:
[INFO]
[INFO] Gluten Parent Pom .................................. SUCCESS [  0.265 s]
[INFO] Gluten UI .......................................... SUCCESS [  8.668 s]
[INFO] Gluten Shims ....................................... SUCCESS [  0.109 s]
[INFO] Gluten Shims Common ................................ SUCCESS [  9.270 s]
[INFO] Gluten Shims for Spark 3.3 ......................... SUCCESS [ 12.868 s]
[INFO] Gluten Core ........................................ FAILURE [ 26.929 s]
```

### Spark version

None

### Spark configurations

_No response_

### System information

_No response_

### Relevant logs

_No response_

--- a/gluten-core/src/main/scala/io/glutenproject/execution/ScanTransformerFactory.scala
+++ b/gluten-core/src/main/scala/io/glutenproject/execution/ScanTransformerFactory.scala
@@ -71,6 +71,7 @@ object ScanTransformerFactory {
       case _ if dataSourceV2TransformerExists(scan.getClass.getName) =>
         val cls = lookupDataSourceV2Transformer(scan.getClass.getName)
         cls
+          .getDeclaredConstructor()
           .newInstance()
           .asInstanceOf[DataSourceV2TransformerRegister]
           .createDataSourceV2Transformer(batchScanExec, newPartitionFilters)

